### PR TITLE
logging: Log only for unhandled errors, remove all the verbose logging.

### PR DESCRIPTION
**Pull request by harshavardhana**
This patch brings in the removal of debug logging altogether, instead
we bring in the functionality of being able to trace the errors properly
pointing back to the origination of the problem.

To enable tracing you need to enable "MINIO_TRACE" set to "1" or "true"
environment variable which would print back traces whenever there is an
error which is unhandled or at the handler layer.

By default this tracing is turned off and only user level logging is provided.

### CODE REVIEW

File: accesslog-handler.go [hunk 1]
```diff
@@ -58,9 +58,9 @@ type LogMessage struct {
 
 func (h *accessLogHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	message, err := getLogMessage(w, req)
```
Comment (abperiasamy) [inline accesslog-handler.go:60]: Access log implementation needs rewrite. Remind me about it.

File: accesslog-handler.go [hunk 2]
```diff
@@ -58,9 +58,9 @@ type LogMessage struct {
...
-	fatalIf(err, "Unable to extract http message.", nil)
+	fatalIf(err, "Unable to extract http message.")
```
Comment (abperiasamy) [inline accesslog-handler.go]: Unable to parse HTTP request and response fields.

File: accesslog-handler.go [hunk 3]
```diff
@@ -58,9 +58,9 @@ type LogMessage struct {
...
 	_, err = h.accessLogFile.Write(message)
-	fatalIf(err, "Writing to log file failed.", nil)
+	fatalIf(err, "Writing to log file failed.")
```
Comment (abperiasamy) [inline accesslog-handler.go]: Unable to log HTTP access.

File: accesslog-handler.go [hunk 4]
```diff
@@ -112,7 +112,7 @@ func getLogMessage(w http.ResponseWriter, req *http.Request) ([]byte, error) {
 // setAccessLogHandler logs requests
 func setAccessLogHandler(h http.Handler) http.Handler {
 	file, err := os.OpenFile("access.log", os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600)
-	fatalIf(err, "Unable to open access log.", nil)
```
Comment (abperiasamy) [inline accesslog-handler.go:115]: Failed to open access log file.

File: bucket-handlers.go [hunk 5]
```diff
@@ -540,7 +540,7 @@ func (api objectAPIHandlers) HeadBucketHandler(w http.ResponseWriter, r *http.Re
 	}
 
 	if _, err := api.ObjectAPI.GetBucketInfo(bucket); err != nil {
-		errorIf(err, "GetBucketInfo failed.", nil)
+		errorIf(err, "Unable to fetch bucket info.")
 		writeErrorResponse(w, r, toAPIErrorCode(err), r.URL.Path)
```
Comment (abperiasamy) [inline bucket-handlers.go:544]: In consistent "." ending. If errorIf doesnt automatically insert it, we should always explicitly add it.

File: bucket-policy-handlers.go [hunk 6]
```diff
@@ -67,7 +67,7 @@ func bucketPolicyActionMatch(action string, statement policyStatement) bool {
 	for _, policyAction := range statement.Actions {
 		// Policy action can be a regex, validate the action with matching string.
 		matched, err := regexp.MatchString(policyAction, action)
-		fatalIf(err, "Invalid pattern, please verify the pattern string.", nil)
+		fatalIf(err, "Invalid pattern, please verify the pattern string.")
```
Comment (abperiasamy) [inline bucket-policy-handlers.go]: "Invalid action \"%s\" in bucket policy."

File: certs.go [hunk 7]
```diff
@@ -45,7 +45,7 @@ func getCertsPath() (string, error) {
 // mustGetCertsPath must get certs path.
 func mustGetCertsPath() string {
 	certsPath, err := getCertsPath()
-	fatalIf(err, "Unable to retrieve certs path.", nil)
+	fatalIf(err, "Unable to retrieve certs path.")
```
Comment (abperiasamy) [inline certs.go]: Failed to get certificate path.

File: config-migrate.go [hunk 8]
```diff
@@ -40,18 +40,18 @@ func purgeV1() {
 	if err != nil && os.IsNotExist(err) {
 		return
 	}
-	fatalIf(err, "Unable to load config version ‘1’.", nil)
+	fatalIf(err, "Unable to load config version ‘1’.")
 
 	if cv1.Version == "1" {
 		console.Println("Unsupported config version ‘1’ found, removed successfully.")
 		/// Purge old fsUsers.json file
```
Comment (abperiasamy) [inline config-migrate.go:47]:
```
    console.Println("Removed unsupported config version 1.")
```

File: config-migrate.go [hunk 9]
```diff
@@ -40,18 +40,18 @@ func purgeV1() {
...
 		configPath, err := getConfigPath()
-		fatalIf(err, "Unable to retrieve config path.", nil)
+		fatalIf(err, "Unable to retrieve config path.")
 
 		configFile := filepath.Join(configPath, "fsUsers.json")
 		os.RemoveAll(configFile)
 	}
-	fatalIf(errors.New(""), "Unexpected version found ‘"+cv1.Version+"’, cannot migrate.", nil)
+	fatalIf(errors.New(""), "Unexpected version found ‘"+cv1.Version+"’, cannot migrate.")
```
Comment (abperiasamy) [inline config-migrate.go]: fatalIf(errors.New(""), "Failed to migrate unrecognized config version \""+cv1.Version+"\".")

File: config-migrate.go [hunk 10]
```diff
@@ -98,14 +98,14 @@ func migrateV2ToV3() {
...
+	fatalIf(err, "Unable to initialize config.")
 
 	configFile, err := getConfigFile()
-	fatalIf(err, "Unable to get config file.", nil)
+	fatalIf(err, "Unable to get config file.")
 
 	// Migrate the config.
 	err = qc.Save(configFile)
-	fatalIf(err, "Migrating from version ‘"+cv2.Version+"’ to ‘"+srvConfig.Version+"’ failed.", nil)
+	fatalIf(err, "Migrating from version ‘"+cv2.Version+"’ to ‘"+srvConfig.Version+"’ failed.")
```
Comment (abperiasamy) [inline config-migrate.go]: fatalIf(err, "Failed to migrate config from version ‘"+cv2.Version+"’ to ‘"+srvConfig.Version+"’.")

File: config-migrate.go [hunk 11]
```diff
@@ -137,12 +137,12 @@ func migrateV3ToV4() {
...
 	qc, err := quick.New(srvConfig)
-	fatalIf(err, "Unable to initialize the quick config.", nil)
+	fatalIf(err, "Unable to initialize the quick config.")
 	configFile, err := getConfigFile()
-	fatalIf(err, "Unable to get config file.", nil)
+	fatalIf(err, "Unable to get config file.")
 
 	err = qc.Save(configFile)
-	fatalIf(err, "Migrating from version ‘"+cv3.Version+"’ to ‘"+srvConfig.Version+"’ failed.", nil)
+	fatalIf(err, "Migrating from version ‘"+cv3.Version+"’ to ‘"+srvConfig.Version+"’ failed.")
```
Comment (abperiasamy) [inline config-migrate.go:146]: same as above.

File: docs/logging.md [hunk 12]
```diff
@@ -1,11 +1,9 @@
...
 
-- `log.Fatalf`
-- `log.Errorf`
-- `log.Warnf`
-- `log.Infof`
+- `fatalIf`
+- `errorIf`
 
-Logging is enabled across the codebase. There are three types of logging supported.
+There are three types of logging supported.
```
Comment (abperiasamy) [inline docs/logging.md]: Supported logging targets:

File: logger-console-hook.go [hunk 13]
```diff
@@ -36,10 +36,11 @@ func enableConsoleLogger() {
 		log.Out = ioutil.Discard
 		return
 	}
+
 	// log.Out and log.Formatter use the default versions.
 	// Only set specific log level.
 	lvl, err := logrus.ParseLevel(clogger.Level)
-	fatalIf(err, "Unknown log level detected, please fix your console logger configuration.", nil)
+	fatalIf(err, "Unknown log level detected, please fix your console logger configuration.")
```
Comment (abperiasamy) [inline logger-console-hook.go]: fatalIf(err, "Unknown log level found in the config file.")

File: logger-file-hook.go [hunk 14]
```diff
@@ -40,13 +40,13 @@ func enableFileLogger() {
...
 	file, err := os.OpenFile(flogger.Filename, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600)
-	fatalIf(err, "Unable to open log file.", nil)
+	fatalIf(err, "Unable to open log file.")
 
 	// Add a local file hook.
 	log.Hooks.Add(&localFile{file})
 
 	lvl, err := logrus.ParseLevel(flogger.Level)
-	fatalIf(err, "Unknown log level detected, please fix your console logger configuration.", nil)
+	fatalIf(err, "Unknown log level detected, please fix your console logger configuration.")
```
Comment (abperiasamy) [inline logger-file-hook.go:50]: same as above.

File: logger-syslog-hook_nix.go [hunk 15]
```diff
@@ -41,11 +41,11 @@ type syslogHook struct {
 // enableSyslogLogger - enable logger at raddr.
 func enableSyslogLogger(raddr string) {
 	syslogHook, err := newSyslog("udp", raddr, syslog.LOG_ERR, "MINIO")
-	fatalIf(err, "Unable to instantiate syslog.", nil)
+	fatalIf(err, "Unable to instantiate syslog.")
```
Comment (abperiasamy) [inline logger-syslog-hook_nix.go:45]: Unable to initialize syslog logger.

File: main.go [hunk 16]
```diff
@@ -195,15 +163,16 @@ func main() {
 		defer profile.Start(profile.BlockProfile, profile.ProfilePath(mustGetProfilePath())).Stop()
 	}
 
+	// Set global trace flag.
+	trace := os.Getenv("MINIO_TRACE")
+	globalTrace = trace == "1" || strings.ToLower(trace) == "true"
+
```
Comment (abperiasamy) [inline main.go:168]: Remove "true" option. Only one way to do one thing.

File: namespace-lock.go [hunk 17]
```diff
@@ -91,10 +90,7 @@ func (n *nsLockMap) unlock(volume, path string, readLock bool) {
 			nsLk.Unlock()
 		}
 		if nsLk.ref == 0 {
-			log.WithFields(logrus.Fields{
-				"volume": volume,
-				"path":   path,
-			}).Error("ref count in NS lock can not be 0.")
+			errorIf(errors.New("Ref count in NS lock cannot be 0."), "Invalid reference count detected.")
```
Comment (abperiasamy) [inline namespace-lock.go]: errorIf(errors.New("Namespace reference count cannot be 0."), "Invalid reference count detected.")

File: object-common-multipart.go [hunk 18]
```diff
@@ -552,9 +536,8 @@ func isUploadIDExists(storage StorageAPI, bucket, object, uploadID string) bool
 		if err == errFileNotFound {
 			return false
 		}
-		log.Errorf("StatFile failed wtih %s", err)
+		errorIf(err, "Unexpected error encountered")
 		return false
```
Comment (abperiasamy) [inline object-common-multipart.go:540]: errorIf(err, "Stat failed on bucket "+minioMetaBucket+":"+"uploadIDPath")

File: object-common.go [hunk 19]
```diff
@@ -268,7 +253,7 @@ func isBucketExist(storage StorageAPI, bucketName string) bool {
 		if err == errVolumeNotFound {
 			return false
 		}
-		log.Errorf("StatVol failed with %s", err)
+		errorIf(err, "Unable to stat volume.")
```
Comment (abperiasamy) [inline object-common.go]: Failed to stat volume %s

> Include bucketname in error message.

File: routers.go [hunk 20]
```diff
@@ -42,11 +42,11 @@ func newObjectLayer(exportPaths ...string) (ObjectLayer, error) {
 // configureServer handler returns final handler for the http server.
 func configureServerHandler(srvCmdConfig serverCmdConfig) http.Handler {
 	objAPI, err := newObjectLayer(srvCmdConfig.exportPaths...)
-	fatalIf(err, "Initializing object layer failed.", nil)
+	fatalIf(err, "Initializing object layer failed.")
```
Comment (abperiasamy) [inline routers.go]: Unable to initialize object layer.

File: routers.go [hunk 21]
```diff
@@ -42,11 +42,11 @@ func newObjectLayer(exportPaths ...string) (ObjectLayer, error) {
...
 
 	// Initialize storage rpc server.
 	storageRPC, err := newRPCServer(srvCmdConfig.exportPaths[0]) // FIXME: should only have one path.
-	fatalIf(err, "Initializing storage rpc server failed.", nil)
+	fatalIf(err, "Initializing storage rpc server failed.")
```
Comment (abperiasamy) [inline routers.go:50]: Unable to initialize storage RPC server.

File: server-main.go [hunk 22]
```diff
@@ -91,14 +91,14 @@ func configureServer(srvCmdConfig serverCmdConfig) *http.Server {
 // getListenIPs - gets all the ips to listen on.
 func getListenIPs(httpServerConf *http.Server) (hosts []string, port string) {
 	host, port, err := net.SplitHostPort(httpServerConf.Addr)
-	fatalIf(err, "Unable to split host port.", nil)
+	fatalIf(err, "Unable to split host port.")
```
Comment (abperiasamy) [inline server-main.go:95]: Unable to parse host port.

File: server-main.go [hunk 23]
```diff
@@ -91,14 +91,14 @@ func configureServer(srvCmdConfig serverCmdConfig) *http.Server {
...
 
 	switch {
 	case host != "":
 		hosts = append(hosts, host)
 	default:
 		addrs, err := net.InterfaceAddrs()
-		fatalIf(err, "Unable to get interface address.", nil)
+		fatalIf(err, "Unable to get interface address.")
 		for _, addr := range addrs {
```
Comment (abperiasamy) [inline server-main.go:102]: Unable to determine network interface addresses.

File: server-main.go [hunk 24]
```diff
@@ -135,10 +135,10 @@ func initServerConfig(c *cli.Context) {
 	// Validate if both keys are specified and they are valid save them.
 	if accessKey != "" && secretKey != "" {
 		if !isValidAccessKey.MatchString(accessKey) {
-			fatalIf(errInvalidArgument, "Access key does not have required length", nil)
+			fatalIf(errInvalidArgument, "Access key does not have required length")
```
Comment (abperiasamy) [inline server-main.go]: Invalid access key.

isValidAccessKey() -> could fail for other reasons as well in the future.

File: server-main.go [hunk 25]
```diff
@@ -135,10 +135,10 @@ func initServerConfig(c *cli.Context) {
...
 		}
 		if !isValidSecretKey.MatchString(secretKey) {
-			fatalIf(errInvalidArgument, "Secret key does not have required length", nil)
+			fatalIf(errInvalidArgument, "Secret key does not have required length")
 		}
```
Comment (abperiasamy) [inline server-main.go:142]: same as above

File: server-main.go [hunk 26]
```diff
@@ -162,9 +162,9 @@ func checkServerSyntax(c *cli.Context) {
 // Extract port number from address address should be of the form host:port.
 func getPort(address string) int {
 	_, portStr, err := net.SplitHostPort(address)
-	fatalIf(err, "Unable to split host port.", nil)
+	fatalIf(err, "Unable to split host port.")
 	portInt, err := strconv.Atoi(portStr)
```
Comment (abperiasamy) [inline server-main.go:166]: Unable to parse host port.

File: server-main.go [hunk 27]
```diff
@@ -201,17 +201,17 @@ func checkPortAvailability(port int) {
...
 		addrs, err := ifc.Addrs()
 		if err != nil {
-			fatalIf(err, fmt.Sprintf("Unable to list addresses on interface %s.", ifc.Name), nil)
+			fatalIf(err, fmt.Sprintf("Unable to list addresses on interface %s.", ifc.Name))
 		}
 		for _, addr := range addrs {
 			ipnet, ok := addr.(*net.IPNet)
 			if !ok {
-				errorIf(errors.New(""), "Interface type assertion to (*net.IPNet) failed.", nil)
+				errorIf(errors.New(""), "Interface type assertion to (*net.IPNet) failed.")
```
Comment (abperiasamy) [inline server-main.go]: Failed to assert type on (*net.IPNet) interface.

File: server-main.go [hunk 28]
```diff
@@ -224,14 +224,14 @@ func checkPortAvailability(port int) {
 			if err != nil {
 				if isAddrInUse(err) {
 					// Fail if port is already in use.
-					fatalIf(err, fmt.Sprintf("Unable to listen on IP %s, port %.d", tcpAddr.IP, tcpAddr.Port), nil)
+					fatalIf(err, fmt.Sprintf("Unable to listen on IP %s, port %.d", tcpAddr.IP, tcpAddr.Port))
 				} else {
```
Comment (abperiasamy) [inline server-main.go:228]: IP:port style is better.

File: server-main.go [hunk 29]
```diff
@@ -224,14 +224,14 @@ func checkPortAvailability(port int) {
...
 					// Ignore other errors.
 					continue
 				}
 			}
 			if err = l.Close(); err != nil {
-				fatalIf(err, fmt.Sprintf("Unable to close listener on IP %s, port %.d", tcpAddr.IP, tcpAddr.Port), nil)
+				fatalIf(err, fmt.Sprintf("Unable to close listener on IP %s, port %.d", tcpAddr.IP, tcpAddr.Port))
```
Comment (abperiasamy) [inline server-main.go]: same as above.

File: server-main.go [hunk 30]
```diff
@@ -316,5 +316,5 @@ func serverMain(c *cli.Context) {
 		// Fallback to http.
 		err = apiServer.ListenAndServe()
 	}
-	errorIf(err, "Failed to start the minio server.", nil)
+	errorIf(err, "Failed to start the minio server.")
```
Comment (abperiasamy) [inline server-main.go]: Failed to start minio server.

File: update-main.go [hunk 31]
```diff
@@ -89,15 +89,15 @@ func (u updateMessage) String() string {
 		return updateMessage("You are already running the most recent version of ‘minio’.")
 	}
 	msg, err := colorizeUpdateMessage(u.Download)
-	fatalIf(err, "Unable to colorize experimental update notification string ‘"+msg+"’.", nil)
+	fatalIf(err, "Unable to colorize experimental update notification string ‘"+msg+"’.")
 	return msg
```
Comment (abperiasamy) [inline update-main.go:93]: Unable to colorize update notice.

File: update-main.go [hunk 32]
```diff
@@ -166,25 +166,25 @@ func getReleaseUpdate(updateURL string, noError bool) updateMessage {
...
 		return updateMsg
 	}
-	fatalIf((err), "Unable to parse version string as time.", nil)
+	fatalIf((err), "Unable to parse version string as time.")
 
 	// Verify if current minio version is zero.
 	if current.IsZero() && !noError {
 		fatalIf((errors.New("")),
-			"Updates not supported for custom builds. Version field is empty. Please download official releases from https://minio.io/#minio", nil)
+			"Updates not supported for custom builds. Version field is empty. Please download official releases from https://minio.io/#minio")
```
Comment (abperiasamy) [inline update-main.go]: Update mechanism is not supported for

Similar error messages should be consistent.

File: update-main.go [hunk 33]
```diff
@@ -194,7 +194,7 @@ func getReleaseUpdate(updateURL string, noError bool) updateMessage {
 			if noError {
 				return updateMsg
 			}
-			fatalIf((errors.New("")), "Update server responsed with "+data.Status, nil)
+			fatalIf((errors.New("")), "Update server responsed with "+data.Status)
```
Comment (abperiasamy) [inline update-main.go]: Failed to retrieve update notice. %s

File: update-main.go [hunk 34]
```diff
@@ -203,19 +203,19 @@ func getReleaseUpdate(updateURL string, noError bool) updateMessage {
 	if err != nil && noError {
 		return updateMsg
 	}
-	fatalIf((err), "Fetching updates failed. Please try again.", nil)
+	fatalIf((err), "Fetching updates failed. Please try again.")
```
Comment (abperiasamy) [inline update-main.go]: Failed to retrieve update notice. Please try again later.

File: update-main.go [hunk 35]
```diff
@@ -203,19 +203,19 @@ func getReleaseUpdate(updateURL string, noError bool) updateMessage {
...
 	}
-	fatalIf(err, "Please report this issue at https://github.com/minio/minio/issues.", nil)
+	fatalIf(err, "Please report this issue at https://github.com/minio/minio/issues.")
 
 	// Verify if the date is not zero.
 	if latest.IsZero() && !noError {
 		fatalIf((errors.New("")),
-			"Unable to validate any update available at this time. Please open an issue at https://github.com/minio/minio/issues", nil)
+			"Unable to validate any update available at this time. Please open an issue at https://github.com/minio/minio/issues")
 	}
```
Comment (abperiasamy) [inline update-main.go:219]: Failed to retrieve update notice. Please try again later. Please report this issue at https://github.com/minio/minio/issues

File: xl-erasure-v1-createfile.go [hunk 36]
```diff
@@ -47,7 +42,7 @@ func (xl XL) cleanupCreateFileOps(volume, path string, writers ...io.WriteCloser
 func closeAndRemoveWriters(writers ...io.WriteCloser) {
 	for _, writer := range writers {
 		if err := safeCloseAndRemove(writer); err != nil {
-			log.Errorf("Closing writer failed with %s", err)
+			errorIf(err, "Unsuccessful close of writer")
```
Comment (abperiasamy) [inline xl-erasure-v1-createfile.go]: Failed to close writer for %s

File: xl-erasure-v1-readfile.go [hunk 37]
```diff
@@ -41,24 +39,15 @@ func (xl XL) ReadFile(volume, path string, startOffset int64) (io.ReadCloser, er
...
 		go func() {
-			if hErr := xl.healFile(volume, path); hErr != nil {
-				log.WithFields(logrus.Fields{
-					"volume": volume,
-					"path":   path,
-				}).Errorf("healFile failed with %s", hErr)
-				return
-			}
+			hErr := xl.healFile(volume, path)
+			errorIf(hErr, "Unable to heal file")
```
Comment (abperiasamy) [inline xl-erasure-v1-readfile.go]: Always include what we are trying to heal. Otherwise, support will be futile.

File: xl-erasure-v1.go [hunk 38]
```diff
@@ -447,20 +424,13 @@ func (xl XL) StatVol(volume string) (volInfo VolInfo, err error) {
...
 		go func() {
-			if hErr := xl.healVolume(volume); hErr != nil {
-				log.WithFields(logrus.Fields{
-					"volume": volume,
-				}).Errorf("healVolume failed with %s", hErr)
-				return
-			}
+			hErr := xl.healVolume(volume)
+			errorIf(hErr, "Unable to heal volume")
 		}()
```
Comment (abperiasamy) [inline xl-erasure-v1.go:434]: same as above.

File: xl-erasure-v1.go [hunk 39]
```diff
@@ -534,23 +504,14 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
...
-			if hErr := xl.healFile(volume, path); hErr != nil {
-				log.WithFields(logrus.Fields{
-					"volume": volume,
-					"path":   path,
-				}).Errorf("healFile failed with %s", hErr)
-				return
-			}
+			hErr := xl.healFile(volume, path)
+			errorIf(hErr, "Unable to heal file")
 		}()
```
Comment (abperiasamy) [inline xl-erasure-v1.go:515]: same as above

File: xl-objects-multipart.go [hunk 40]
```diff
@@ -223,9 +223,7 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 			src := path.Join(mpartMetaPrefix, bucket, object, uploadID, partSuffix)
 			dst := path.Join(mpartMetaPrefix, bucket, object, uploadID, partNumToPartFileName(part.PartNumber))
 			errs[index] = xl.storage.RenameFile(minioMetaBucket, src, minioMetaBucket, dst)
-			if errs[index] != nil {
-				log.Errorf("Unable to rename file %s to %s, failed with %s", src, dst, errs[index])
-			}
+			errorIf(errs[index], "Unable to rename file")
```
Comment (abperiasamy) [inline xl-objects-multipart.go]: same as above.

File: xl-objects.go [hunk 41]
```diff
@@ -49,20 +49,17 @@ func isValidFormat(storage StorageAPI, exportPaths ...string) bool {
 	// Load saved XL format.json and validate.
 	xl, err := loadFormatXL(storage)
 	if err != nil {
-		log.Errorf("loadFormatXL failed with %s", err)
+		errorIf(err, "Unable to load XL format")
```
Comment (abperiasamy) [inline xl-objects.go]: Unable to load format file %s

File: xl-objects.go [hunk 42]
```diff
@@ -90,11 +86,9 @@ func newXLObjects(exportPaths ...string) (ObjectLayer, error) {
 				Disks:   exportPaths,
 			})
 			if errSave != nil {
-				log.Errorf("saveFormatXL failed with %s", errSave)
 				return nil, errSave
 			}
 		} else {
-			log.Errorf("Unable to check backend format %s", err)
 			if err == errReadQuorum {
 				errMsg := fmt.Sprintf("Not all disks %s are available, did not meet read quroum.", exportPaths)
```
Comment (abperiasamy) [inline xl-objects.go]: Disks %s are offline.  Unable to establish quorum

File: accesslog-handler.go (hunk 1 above)
Comment (adriancooney) [reply accesslog-handler.go:60]: Reminding! Looking for this feature.

Comment (krishnasrinivas) [reply accesslog-handler.go:60]: An issue was opened on this: https://github.com/minio/minio/issues/2202

@adriancooney you can run github.com/krishnasrinivas/httpdump proxy in front of minio for request/response dump.

**Pull request merged**